### ISSUE TITLE
Gegno Mining Overhaul

### ISSUE THREAD

**Issue by Saugia** [endless-sky/endless-sky#7936]
### Problem Description

The most important aspect of Gegno lore is mining, as the entire conflict is based around the ability to and the conflict in important mining systems, mainly the system Vesvii, home to an enormous ringworld.

Due to their low tier as well as their lore basis, the Gegno are planned to have the best mining equipment in the game, and likely some unique mining outfits. Currently, as they were recently introduced, their mining aspects are very limited, as the only actual mining interaction is in a single ship class they own with a very powerful mining drill. I'd like to expand upon this much more in the coming future, but it will require new mechanics not part of the game quite yet.

Most importantly, this is pending on @Amazinite 's new mining overhaul. So I've created this issue to keep track of relative additions to the game that will help make the Gegno more mining-themed. 

### Related Issue Links

I will add related links here when appropriate PRs and Issues are raised.

### Desired Solution

The desired solution to expanding upon the Gegno's mining aspects are listed as follows.

- [ ] Introduce the planned Mining Overhaul.  Importantly, percent yield, and allowing Gegno mining weapons to succeed in that.
- [ ] Introduce a unique mining mechanic. This could be refinery internals, specialized ore cargo bays, etc..
- [ ] Introduce unique mining gameplay aspects. These could be larger asteroids, new materials, etc..
- [ ] Introduce new Gegno ship classes to broaden their mining shipyard. Light Freighters, alternate mining ships, etc..
- [ ] Introduce new Gegno mining weapons. Different types of drills, perhaps even Scin pulse laser drills in the far future.
- [ ] 

### Alternative Approaches

No alternate approaches, as the above is either planned or plausible.

### Additional Context

The Gegno's Introduction: #661

**Comment by Zitchas** [comment]
I might suggest a few additions as beneficial to the mining related stuff:
- https://github.com/endless-sky/endless-sky/pull/6669
- https://github.com/endless-sky/endless-sky/pull/6726 (this one is more general fighter improvements, but likely to be highly beneficial to low tier fighters even more so than other areas)
- https://github.com/endless-sky/endless-sky/pull/7103
